examples/sampling/pyg/node_classification.py

```diff
@@ -0,0 +1,260 @@
+
+
+class GraphSAGE(torch.nn.Module):
+    #####################################################################
+    # (HIGHLIGHT) Define the GraphSAGE model architecture.
+    #
+    # - This class inherits from `torch.nn.Module`.
+    # - Two convolutional layers are created using the SAGEConv class from PyG.
+    # - 'in_size', 'hidden_size', 'out_size' are the sizes of
+    #   the input, hidden, and output features, respectively.
+    # - The forward method defines the computation performed at every call.
+    #####################################################################
+    def __init__(self, in_size, hidden_size, out_size):
+        super(GraphSAGE, self).__init__()
+        self.layers = torch.nn.ModuleList()
+        self.layers.append(SAGEConv(in_size, hidden_size))
+        self.layers.append(SAGEConv(hidden_size, hidden_size))
+        self.layers.append(SAGEConv(hidden_size, out_size))
+
+    def forward(self, blocks, x, device):
+        h = x
+        for i, (layer, block) in enumerate(zip(self.layers, blocks)):
+            # print(f"Layer {i}: Feature tensor shape before convolution: {h.shape}")
+            src, dst = block.edges()
+            edge_index = torch.stack([src, dst], dim=0)
+            h = layer(h, edge_index)
+            # print(f"Layer {i}: Feature tensor shape after convolution: {h.shape}")
+            if i != len(blocks) - 1:
+                h = F.relu(h)
+            # print(f"Layer {i}: Number of destination nodes: {block.number_of_dst_nodes()}")
+
+            h = h[: block.number_of_dst_nodes()]
+            # print(f"Layer {i}: Feature tensor shape after slicing: {h.shape}")
+
+        return h
+
+
+def create_dataloader(dataset_set, graph, feature, device, is_train):
+    #####################################################################
+    # (HIGHLIGHT) Create a data loader for efficiently loading graph data.
+    #
+    # - 'ItemSampler' samples mini-batches of node IDs from the dataset.
+    # - 'sample_neighbor' performs neighbor sampling on the graph.
+    # - 'FeatureFetcher' fetches node features based on the sampled subgraph.
+    # - 'CopyTo' copies the fetched data to the specified device.
+
+    #####################################################################
+    # Create a datapipe for mini-batch sampling with a specific neighbor fanout.
+    # Here, [10, 10, 10] specifies the number of neighbors sampled for each node at each layer.
+    # We're using `sample_neighbor` for consistency with DGL's sampling API.
+    # Note: GraphBolt offers additional sampling methods, such as `sample_layer_neighbor`,
+    # which could provide further optimization and efficiency for GNN training.
+    # Users are encouraged to explore these advanced features for potentially improved performance.
+
+    # Initialize an ItemSampler to sample mini-batches from the dataset.
+    datapipe = gb.ItemSampler(
+        dataset_set, batch_size=1024, shuffle=is_train, drop_last=not is_train
```

Comment:
```suggestion
        dataset_set, batch_size=1024, shuffle=is_train, drop_last=is_train
```
Sorry for the mistake, when we are training, we need to drop the last minibatch that does not have batch_size items, however when validating and testing, we should keep the last minibatch.